src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1157,6 +1160,121 @@ static bool isMaybeDefaultValue(TypeWithState valueType)
             }
         }
 
+        internal static bool AreParameterAnnotationsCompatible(
```

Comment:
Perhaps this should be named `CheckOverridingParameterAnnotations`? Since we tend to "blame" the overriding method for not conforming to the requirements of the overridden method? Also wondering if it might make sense to declare closer to its use site in SourceMemberContainerSymbol_ImplementationChecks.

Reply:
I initially has this method in `SourceMemberContainerSymbol_ImplementationChecks`, but that required to expose multiple private methods of `NulllableWalker` as internal. So I figured this was a more natural place.
I'll adjust the name

Reply:
I ended up keeping the current name because this method doesn't report any diagnostics. The caller is assigning blame to the overriding/implementing parameter ;-)

---
In reply to: [377404557](https://github.com/dotnet/roslyn/pull/41336#discussion_r377404557) [](ancestors = 377404557)